scalding-core/src/test/scala/com/twitter/scalding/JoinerFlowProcessTest.scala

```diff
@@ -0,0 +1,95 @@
+package com.twitter.scalding
+
+import cascading.flow.FlowException
+import cascading.pipe.CoGroup
+import cascading.pipe.joiner.{ JoinerClosure, InnerJoin }
+import cascading.tuple.Tuple
+import org.scalatest.{ Matchers, WordSpec }
+
+import java.util.{ Iterator => JIterator }
+
+class CheckFlowProcessJoiner(uniqueID: UniqueID) extends InnerJoin {
+  override def getIterator(joinerClosure: JoinerClosure): JIterator[Tuple] = {
+    val flowProcess = RuntimeStats.getFlowProcessForUniqueId(uniqueID)
+    if (flowProcess == null) {
+      throw new NullPointerException("No active FlowProcess was available.")
+    }
+
+    super.getIterator(joinerClosure)
+  }
+}
+
+class TestWrappedJoinerJob(args: Args) extends Job(args) {
+  val uniqueID = UniqueID.getIDFor(flowDef)
+
+  val inA = Tsv("inputA", ('a, 'b))
+  val inB = Tsv("inputA", ('x, 'y))
+
+  val joiner = {
+    val checkJoiner = new CheckFlowProcessJoiner(uniqueID)
+    if (args.boolean("wrapJoiner")) WrappedJoiner(checkJoiner) else checkJoiner
+  }
+
+  val p1 = new CoGroup(inA, 'a, inB, 'x, joiner)
+
+  // The .forceToDisk is necessary to have the test work properly.
+  p1.forceToDisk.write(Tsv("output"))
+}
+
+class CheckForFlowProcessInOrdinaryJob(args: Args) extends Job(args) {
+  val uniqueID = UniqueID.getIDFor(flowDef)
+
+  val inA = Tsv("inputA", ('a, 'b))
+  val inB = Tsv("inputA", ('x, 'y))
+
+  val joiner = new CheckFlowProcessJoiner(uniqueID)
+
+  // The .forceToDisk is necessary to have the test work properly.
+  inA.joinWithSmaller('a -> 'x, inB, joiner = joiner).forceToDisk.write(Tsv("output"))
+}
+
```

Comment:
can you also add a test for the Typed-API? We recommend that one now (and it uses a different code path than the field API for joins).

To test, you should be able to add a stat in a `.reduce` following a join in the Typed-API.